src/read.c

```diff
@@ -501,16 +501,18 @@ void ReadCallVarAss (
     if (WarnOnUnboundGlobalsRNam == 0)
       WarnOnUnboundGlobalsRNam = RNamName("WarnOnUnboundGlobals");
 
-    if ( type == 'g'
-      && STATE(CountNams) != 0
-      && var != STATE(CurrLHSGVar)
-      && VAL_GVAR(var) == 0
-      && ExprGVar(var) == 0
-      && ! STATE(IntrIgnoring)
-      && ! GlobalComesFromEnclosingForLoop(var)
-      && (GAPInfo == 0 || !IS_REC(GAPInfo) || !ISB_REC(GAPInfo,WarnOnUnboundGlobalsRNam) ||
-             ELM_REC(GAPInfo,WarnOnUnboundGlobalsRNam) != False )
-      && ! SyCompilePlease )
+    if ( type == 'g'                /* Reading a global variable          */
```

Comment:
How about using single-line comments `// Reading ...` for these new comments? Now that we agreed C99 is fine... :-). Not that this is important in any way